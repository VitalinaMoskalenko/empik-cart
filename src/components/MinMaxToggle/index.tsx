import React, { useState } from "react";
import styled from "styled-components";
import { SmallBody } from "../Paragraphs";

type PropsType = {
  min: number;
  max: number;
  isBlocked?: boolean;
  onValueChanged: (value: number) => void;
};

const Container = styled.div`
  display: flex;
  flex-direction: row;
`;

const Button = styled.button`
  width: 40px;
  height: 40px;
  padding: 10px;
`;

const Count = styled(SmallBody)`
  text-align: center;
  width: 20px;
  padding: 10px;
`;

const MinMaxToggle = ({ min, max, isBlocked, onValueChanged }: PropsType) => {
  const [count, setCount] = useState(1);

  const onMinusClick = () => {
    if (min < count) {
      onValueChanged(count - 1);
      setCount(count - 1);
    }
  };

  const onPlusClick = () => {
    if (max > count) {
      onValueChanged(count + 1);
      setCount(count + 1);
    }
  };

  return (
    <Container>
      <Button onClick={onMinusClick} disabled={isBlocked}>
        <SmallBody>-</SmallBody>
      </Button>
      <Count>{count}</Count>
      <Button onClick={onPlusClick} disabled={isBlocked}>
        <SmallBody>+</SmallBody>
      </Button>
    </Container>
  );
};

export default MinMaxToggle;
