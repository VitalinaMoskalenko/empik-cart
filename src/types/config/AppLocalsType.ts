export enum AppLocalsType {
  pl = "pl",
}
