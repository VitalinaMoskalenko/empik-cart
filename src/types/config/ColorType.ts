export enum ColorType {
  white = "white",
  black = "black",
  grey = "grey",
  orange = "#ff6903",
}
