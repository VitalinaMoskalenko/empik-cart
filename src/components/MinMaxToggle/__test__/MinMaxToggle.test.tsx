import React from "react";
import renderer from "react-test-renderer";
import { ThemeProvider } from "styled-components";
import MinMaxToggle from "../";
import { theme } from "../../../config";

it("renders MinMaxToggle correctly", () => {
  const minMaxToggleComponent = renderer
    .create(
      <ThemeProvider theme={theme}>
        <MinMaxToggle min={1} max={1} onValueChanged={() => {}} />
      </ThemeProvider>
    )
    .toJSON();

  expect(minMaxToggleComponent).toMatchSnapshot();
});
