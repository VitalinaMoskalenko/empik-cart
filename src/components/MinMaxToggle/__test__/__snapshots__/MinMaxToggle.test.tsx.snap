// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MinMaxToggle correctly 1`] = `
<div
  className="sc-dlnjwi kcZDqU"
>
  <button
    className="sc-hKFxyN fyweba"
    onClick={[Function]}
  >
    <p
      className="sc-gtsrHT ckjQab"
    >
      -
    </p>
  </button>
  <p
    className="sc-gtsrHT sc-eCApnc ckjQab dWxdxR"
  >
    1
  </p>
  <button
    className="sc-hKFxyN fyweba"
    onClick={[Function]}
  >
    <p
      className="sc-gtsrHT ckjQab"
    >
      +
    </p>
  </button>
</div>
`;
